libvast/vast/hash/xxhash.hpp

```diff
@@ -30,10 +30,6 @@ static constexpr int xxh_force_memory_access = XXH_FORCE_MEMORY_ACCESS;
 static constexpr int xxh_force_memory_access = 0;
 #endif
 
-/// Allow null pointer input when hashing data of length greater 0.
-static constexpr bool xxh_accept_null_input_pointer
-  = XXH_ACCEPT_NULL_INPUT_POINTER;
```

Comment:
Can we do something like this?
```cpp
#ifdef XXHASH < 0.8.1
#define XXH_ACCEPT_NULL_INPUT_POINTER
#endif
```

Reply:
I don't think this changes the logic at all, does it?

Reply:
You need to define the constant, actually prior to including the xxHash header, as it changes how null pointers are handled. If you do not define this constant, we may get a crash if we attempt to hash `nullptr`, even if the size parameter is 0.

Reply:
In that case we should let CMake define it. That'd be the proper way to do it.

Reply:
Done in CMake in a separate commit.